emission/tests/modellingTests/TestGreedySimilarityBinning.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+import emission.analysis.modelling.trip_model.greedy_similarity_binning as eamtg
+import emission.tests.modellingTests.modellingTestAssets as etmm
+import emission.analysis.modelling.similarity.od_similarity as eamso
+import json
+import logging
+
+
+class TestGreedySimilarityBinning(unittest.TestCase):
+
+    def setUp(self) -> None:
+        logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s',
+        level=logging.DEBUG)
+
+    def testBinning(self):
+        """
+        when $should_be_grouped trips are the same, they should appear in a bin
+        """
+        label_data = {
+            "mode_confirm": ['walk', 'bike', 'transit'],
+            "purpose_confirm": ['work', 'home', 'school'],
+            "replaced_mode": ['drive']
+        }
+
+        n = 20
+        should_be_grouped = 5
+        trips = etmm.generate_mock_trips(
+            user_id="joe", 
+            trips=n, 
+            origin=(0, 0), 
+            destination=(1, 1), 
+            label_data=label_data, 
+            within_threshold=should_be_grouped, 
+            threshold=0.001,  # ~ 111 meters in degrees WGS84
+        )
+
+        model_config = {
+            "metric": "od_similarity",
+            "similarity_threshold_meters": 500,      # meters,
+            "apply_cutoff": False,
+            "incremental_evaluation": False
+        }
+        model = eamtg.GreedySimilarityBinning(model_config)
+        
+        model.fit(trips)
+
+        # $should_be_grouped trip features should appear together in one bin
+        at_least_one_large_bin = any(map(lambda b: len(b['features']) >= should_be_grouped, model.bins.values()))
```

Comment:
why is this `>=` instead of `=`? It looks like `testPrediction` assumes that if we generate trips within a threshold they will all end up in the same bin...

Reply:
you're right. fixed.